### PR TITLE
Update dependencies and version

### PR DESCRIPTION
**Pull request by erikmannerfelt**
* All dependencies are now on conda-forge!
* Removed `richdem` from the `environment.yml` as it is an optional dependency
* Removed `pdal` as it is no longer in use.
* Set minimum python version to 3.7 (because of type annotations)
* Ran `black` on `setup.py` since formatting was all over the place.
* The `build` workflow no longer uses pip to install dependencies.
* Prepared `build` workflow to build on Windows and MacOS (didn't enable it yet).
* Updated `README.md` accordingly.

In 0c3c06f, I also added:
* Documentation versioning now dynamically follows the setup.py version (it was hardcoded before..)
* Added manual dispatch to PyPi publishing workflow (see https://github.com/GlacioHack/GeoUtils/pull/192)
* Bumped version to `v0.0.3` (bugfixes on all fronts)

In aa3485f, I added a `dev-environment.yml`, making #123  obsolete.

In ccb4885 and 82b702b:
* Added missing dev requirement
* Fixed test in `test_coreg.py` that sometimes fails randomly!
* Simplified doc building test, making it easier to read, and easier to work with on Windows.

### CODE REVIEW

File: .github/workflows/python-package.yml
Comment (adehecq) [inline .github/workflows/python-package.yml:46]: Is it better to have it hard-coded here rather than in an environment.yml?

File: docs/source/conf.py
Comment (adehecq) [inline docs/source/conf.py:28]: Nice!

File: setup.py
Comment (adehecq) [inline setup.py:8]: I see this is hard-coded here. So does it have to be updated each time? Or where should we modify the version number?

File: setup.py
Comment (adehecq) [inline setup.py:32]: Ideally this would also be an optional dependency as most users will be not necessarily be interested in the spstats part... I think it adds up a lot of additional dependencies no?

File: setup.py
Comment (erikmannerfelt) [reply setup.py:8]: Yes, and there's no apparent way to fix this. It's here and in the GitHub release part. Everything else is propagated from them.

**Comment by adehecq** [comment]
Great!
The github parts are still a mystery to me, so I trust you on this!
I made some comments, but feel free to report them as issue or keep them in mind for later if you need to merge this PR.

**Review by adehecq — DISMISSED** [review]

File: setup.py
Comment (erikmannerfelt) [reply setup.py:32]: In its current state, the spstats dependencies are required. We could discuss in the future if that should be the case (@rhugonnet), but for now, this is how it should be!

File: .github/workflows/python-package.yml
Comment (erikmannerfelt) [reply .github/workflows/python-package.yml]: Added a `dev-environment.yml` in aa3485f!

**Comment by erikmannerfelt** [comment]
I just saw that the documentation doesn't build now on readthedocs. This needs to be fixed before merging

**Comment by erikmannerfelt** [comment]
readthedocs is happy again! @adehecq

**Comment by erikmannerfelt** [comment]
I'll merge this as it was approved by @adehecq and I had to dismiss his approval as readthedocs started complaining. But now that's fixed!

**Pull request merged**
